frontend/pages/index.tsx

```diff
@@ -13,7 +17,27 @@ import ExternalLink from "@/components/ExternalLink";
  * The component for the home page
  */
 const HomePage = () => {
+  const router = useRouter();
   const { username } = useAuthContext();
+  const [ email, setEmail ] = useState("");
+
+  const handleEmailChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setEmail(event.target.value);
+  }
+
+  const handleSubmit = async (event: React.ChangeEvent<HTMLFormElement>) => {
+    event.preventDefault();
+    
+    const waitlistEntryData = { email };
```

Comment:
You could remove `waitlistEntryData` and instead pass `email` directly to the request function.

Reply:
A good point. I changed it